Riot/Modules/Onboarding/OnboardingCoordinator.swift

```diff
@@ -191,23 +198,28 @@ final class OnboardingCoordinator: NSObject, OnboardingCoordinatorProtocol {
     // MARK: - Authentication
     
     /// Show the authentication flow, starting at the specified initial screen.
-    private func beginAuthentication(with initialScreen: AuthenticationCoordinator.EntryPoint, onStart: @escaping () -> Void) {
+    private func beginAuthentication(with initialScreen: AuthenticationCoordinator.EntryPoint, onStart: (() -> Void)? = nil) {
         MXLog.debug("[OnboardingCoordinator] beginAuthentication")
         
         let parameters = AuthenticationCoordinatorParameters(navigationRouter: navigationRouter,
                                                              initialScreen: initialScreen,
-                                                             canPresentAdditionalScreens: false)
+                                                             canPresentAdditionalScreens: false,
+                                                             softLogoutCredentials: self.parameters.softLogoutCredentials)
         let coordinator = AuthenticationCoordinator(parameters: parameters)
         coordinator.callback = { [weak self, weak coordinator] result in
             guard let self = self, let coordinator = coordinator else { return }
             
             switch result {
             case .didStart:
-                onStart()
+                onStart?()
             case .didLogin(let session, let authenticationFlow, let authenticationType):
                 self.authenticationCoordinator(coordinator, didLoginWith: session, and: authenticationFlow, using: authenticationType)
             case .didComplete:
                 self.authenticationCoordinatorDidComplete(coordinator)
+            case .clearAllData:
+                self.isShowingLegacyAuthentication = false
+                self.authenticationFinished = false
+                AppDelegate.theDelegate().logoutSendingRequestServer(true, completion: nil)
```

Comment:
Should the completion of this call `cancelAuthentication(flow: .login)` to reset back to the start?

Reply:
Nope, the method calls `masterTabBarController.showOnboardingFlow` at the end.

Reply:
~~Nope, the method calls `masterTabBarController.showOnboardingFlow` at the end.~~

It was a wiser choice to implement like this because `MasterTabbarController showOnboardingFlow` did nothing if already presenting an onboarding.